frontend/src/components/assessments/UpcomingSession.tsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import {
+  Divider,
+  HStack,
+  Spacer,
+  Tag,
+  TagLabel,
+  TagLeftIcon,
+  Text,
+  Tooltip,
+  useDisclosure,
+  VStack,
+} from "@chakra-ui/react";
+
+import { BookIcon } from "../../assets/icons";
+import { formatDate } from "../../utils/GeneralUtils";
+import Copyable from "../common/Copyable";
+import Popover from "../common/Popover";
+import PopoverButton from "../common/PopoverButton";
+
+export type UpcomingSessionProps = {
+  classroomName: string;
+  sessionName: string;
+  // Target date should be the start date of the session UNLESS
+  // the session is active, in which case it should be the end date
+  targetDate: Date;
+  accessCode: string;
+  status: "active" | "upcoming" | "past";
+};
+
+const STATUS_LABELS = {
+  active: "Until",
+  upcoming: "Scheduled",
+  past: "Assigned",
+};
+
+const UpcomingSession = ({
+  classroomName,
+  sessionName,
+  targetDate,
+  accessCode,
+  status,
+}: UpcomingSessionProps): React.ReactElement => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  return (
+    <HStack gap={2} pb={8}>
+      <Tooltip
+        bg="blue.300"
+        borderRadius={4}
+        hasArrow
+        label={classroomName}
+        p={2}
+        placement="left"
+      >
+        <Tag
+          bg="blue.50"
+          borderRadius="full"
+          overflow="hidden"
+          size="lg"
+          w={36}
+        >
+          <TagLeftIcon as={BookIcon} color="blue.300" />
```

Comment:
I think the icon for me is showing as black instead of `blue.300`? not sure if it's the same for you

Reply:
thought I fixed that 💀